src/Extensions/Hosting.Services.Web/ServiceCollectionExtensions.cs

```diff
@@ -15,7 +19,22 @@ public static class ServiceCollectionExtensions
 		/// </summary>
 		public static IServiceCollection AddOmexMiddleware(this IServiceCollection services)
 		{
+			services.TryAddSingleton<ISystemClock, SystemClock>();
+			services.TryAddSingleton<IUserIdentityProvider, IpBasedUserIdentityProvider>();
```

Comment:
I suppose we'll need a chain of identity providers here, or an identity provider with some fallback logic. It might be worth adding more specific interfaces like `IRotatingSaltProvider` or another way to use multiple providers from the beginning.

Reply:
I've added the ability of chainging `IUserIdentityProvider`'s but I don't think it would be useful for salt providers

Reply:
Oh right, I meant `IIpBasedUserIdentityProvider`. But I suppose we can tackle it later when it's needed